pkg/daemon/ceph/client/mirror.go

```diff
@@ -168,3 +176,60 @@ func GetPoolMirroringInfo(context *clusterd.Context, clusterInfo *ClusterInfo, p
 
 	return &poolMirroringInfo, nil
 }
+
+// enableSnapshotSchedule configures the snapshots schedule on a mirrored pool
+func enableSnapshotSchedule(context *clusterd.Context, clusterInfo *ClusterInfo, snapSpec cephv1.SnapshotScheduleSpec, poolName string) error {
+	logger.Infof("enabling snapshot schedule for pool %q", poolName)
+
+	// Build command
+	args := []string{"mirror", "snapshot", "schedule", "--pool", poolName, snapSpec.Interval}
+
+	// If a start time is defined let's add it
+	if snapSpec.StartTime != "" {
+		args = append(args, snapSpec.StartTime)
+	}
+	cmd := NewRBDCommand(context, clusterInfo, args)
+
+	// Run command
+	buf, err := cmd.Run()
+	if err != nil {
+		return errors.Wrapf(err, "failed to enable snapshot schedule on pool %q. %s", poolName, string(buf))
+	}
+
+	logger.Infof("successfully enabled snapshot schedule for pool %q every %q", poolName, snapSpec.Interval)
+	return nil
+}
+
+func enableSnapshotSchedules(context *clusterd.Context, clusterInfo *ClusterInfo, poolSpec cephv1.PoolSpec, poolName string) error {
```

Comment:
If schedules are removed from the list, do we disable them? Or if an existing rule has an updated time applied, will the existing rule be updated, or will it just be added as another schedule? To keep it simple, I wonder if we just need to clear the rules during every reconcile and re-create, unless that messes with the existing snapshots.

Reply:
Today, none of that is done. But it is identical in various places such as the manager modules, pool parameters etc etc. Schedules have no name that we could use to identify themselves. They seem to be identified by their Internal and StartTime properties.

So today, if an interval/starttime is modified we will add a new one without suppressing the old one.
Also, removing one from the list won't remove the snapshot.

All of that can be done though.

Reply:
I think my thoughts for what seems best align with what I read of Travis' thoughts. It seems like treating the manifest as the list of desired schedules is most k-native to me. Which might mean removing from Ceph any non-matching schedules and creating any schedules Ceph is missing.

Reply:
@BlaineEXE @travisn rework as requested and verified. Ready for final review.